docs/rfc/active/000-datadoc-entity/datadoc-entity-rfc.md

```diff
@@ -0,0 +1,72 @@
+- Start Date: (fill me in with today's date, 2022-02-22)
+- RFC PR: https://github.com/linkedin/datahub/pull/4237 
+- Discussion Issue: (GitHub issue this was discussed in before the RFC, if any)
+- Implementation PR(s): (leave this empty)
+
+# Extend data model to model Notebook entity 
+
+## Background
+[Querybook](https://www.querybook.org/) is Pinterest’s open-source big data IDE via a notebook interface. 
+We(Included Health) leverage it as our main querying tool. It has a feature, DataDoc, which organizes rich text, 
+queries, and charts into a notebook to easily document analyses. People could work collaboratively with others in a 
+DataDoc and get real-time updates. We believe it would be valuable to ingest the DataDoc metadata to Datahub and make
+it easily searchable and discoverable by others.
+
+## Summary
+This RFC proposes the data model used to model DataDoc entity. It does not talk about any architecture, API or other 
+implementation details. This RFC only includes minimum data model which could meet our initial goal. If the community 
+decides to adopt this new entity, further effort is needed.
+
+## Detailed design
+
+### DataDoc Model
+![DataDoc High Level Model](DataDoc-high-level-model.png) 
+
+As shown in the above diagram, DataDoc is a document which contains a list of DataDoc cells. It organizes rich text,
+queries, and charts into a notebook to easily document analyses. We could see that the DataDoc model is very similar as 
+Notebook. DataDoc would be viewed as a subset of Notebook. Therefore we are going to model Notebook rather than DataDoc. 
+We will include "subTypes" aspect to differentiate Notebook and DataDoc 
+
+### Notebook Data Model
```

Comment:
One thing that seems to be missing is lineage information. Though storing information about notebook contain may help with discovery among datadocs, I think we need some level of lineage between docs and datasets. Regarding this 1) Is this something we can fetch from QueryBooks or Jupyter notebooks? 2) How would we store information? Ideally, think it should be per cell (specifically query cell and chart cell), but up for debate.

Reply:
I think it makes sense to include lineage information, like linking the notebook to some datasets.

For querybook and jupyter notebooks, I feel we could only extract the lineage information from the query. 

Regarding it's per cell or per notebook, I feel it fits the reality better to make it per cell. But we model the cell as one aspect of a notebook entity. I think it will be better to make it per data entity which is per notebook entity